httpcache.go

```diff
@@ -186,70 +199,67 @@ func (t *Transport) RoundTrip(req *http.Request) (resp *http.Response, err error
 		}
 
 		resp, err = transport.RoundTrip(req)
-		if err == nil && req.Method == "GET" && resp.StatusCode == http.StatusNotModified {
+		if err != nil {
+			if cachedResp != nil && canStaleOnError(req, cachedResp.Header) {
+				// In case of transport failure and stale-if-error activated, returns cached content
+				// when available
+				return cachedResp, nil
+			}
+			t.Cache.Delete(cacheKey)
+			return resp, err
+		}
+
+		if req.Method == "GET" && resp.StatusCode == http.StatusNotModified {
 			// Replace the 304 response with the one from cache, but update with some new headers
 			endToEndHeaders := getEndToEndHeaders(resp.Header)
 			for _, header := range endToEndHeaders {
 				cachedResp.Header[header] = resp.Header[header]
 			}
 			resp = cachedResp
-		} else if (err != nil || (cachedResp != nil && resp.StatusCode >= 500)) &&
-			req.Method == "GET" && canStaleOnError(cachedResp.Header, req.Header) {
+		} else if cachedResp != nil && resp.StatusCode >= 500 && canStaleOnError(req, cachedResp.Header) {
```

Comment:
same question as above.

Reply:
> same question as above.

Good catch.  Fixed in 55ebf79 -> 9bccf17.